presto-hive/src/main/java/com/facebook/presto/hive/RetryDriver.java

```diff
@@ -155,9 +158,19 @@ public RetryDriver stopOnIllegalExceptions()
                 }
                 catch (InterruptedException ie) {
                     Thread.currentThread().interrupt();
-                    throw new RuntimeException(ie);
+                    throw addSuppressedExceptions(new RuntimeException(ie), suppressedExceptions);
                 }
             }
         }
     }
+
+    private static Exception addSuppressedExceptions(Exception exception, List<Throwable> suppressedExceptions)
+    {
+        for (Throwable suppressedException : suppressedExceptions) {
+            if (exception != suppressedException) {
+                exception.addSuppressed(suppressedException);
+            }
+        }
+        return exception;
```

Comment:
This return makes calling code more fluent, but i don't find it more readable. 
What about making this method `void`?